ui/src/app/views/account-creation-success/component.tsx

```diff
@@ -144,7 +152,9 @@ export class AccountCreationSuccessComponent implements DoCheck, OnInit {
     ReactDOM.render(<AccountCreationSuccessReact
         contactEmailOnCreation={this.contactEmail}
         username={this.account.profile.username}
-        creationNonce = {this.account.profile.creationNonce}/>,
+        creationNonce={this.account.profile.creationNonce}
+        profileApi={this.profileApi}
```

Comment:
Rather than passing this through props, could we just import and use it directly? Mocking could be done at the level of `fetch` via e.g. `fetch-mock`

Reply:
So first, the goal here:

Our pattern so far in tests involving API calls has been to use fakes, which in terms of this code means we'd be needing to supply a different implementation of `profileApi`. From my experience I have a strong preference for this approach over using HTTP mocking level to simulate what individual calls to `getProfile` might return at the JSON level. That encoding is a detail of the swagger spec. I think this point is entirely language independent and unrelated to React vs Angular.

The end goal here would be that in my test setup, I can wireup a fake `profileApi` implementation with the appropriate stubbed data, and somehow pass that into the component under test (ideally not staticly, which generally leads to spaghetti code).

Next, the implementation of this in React:

I'm not sure of the most natural approach. From what I've seen, there's `props`, something called the `Context` API (I can't tell whether this is recommended for use), and then a 3rd option to roll something custom (effectively static, but we have some capability to reliably stub things out in a library we control).